packages/gestalt/src/Tooltip.js

```diff
@@ -3,62 +3,128 @@
 import * as React from 'react';
 import Controller from './Controller.js';
 import Text from './Text.js';
+import Link from './Link.js';
 import Box from './Box.js';
 
 const noop = () => {};
 
 type Props = {|
   children: React.Node,
+  href?: string,
+  hrefText?: string,
   inline?: boolean,
   text: string,
 |};
 
-type State = {|
-  hovered: boolean,
-|};
+const initialState = { hoveredIcon: false, hoveredText: false, isOpen: false };
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'hoverInIcon':
+      return {
+        ...state,
+        hoveredIcon: true,
+        isOpen: true,
+      };
+    case 'hoverInText':
+      return {
+        ...state,
+        hoveredText: true,
+        isOpen: true,
+      };
+    case 'hoverOutIcon':
+      return {
+        ...state,
+        hoveredIcon: false,
+        ...(!state.hoveredText ? { isOpen: false } : {}),
+      };
+    case 'hoverOutText':
+      return {
+        ...state,
+        hoveredText: false,
+        ...(!state.hoveredIcon ? { isOpen: false } : {}),
+      };
+    default:
+      throw new Error();
+  }
+};
 
-export default class Tooltip extends React.Component<Props, State> {
-  state = {
-    hovered: false,
+export default function Tooltip({ children, href, hrefText, inline, text }) {
+  const [state, dispatch] = React.useReducer(reducer, initialState);
+  const { isOpen } = state;
+
+  const childRef = React.useRef<?HTMLDivElement>(null);
+  const { current: anchor } = childRef;
+
+  const handleIconMouseEnter = () => {
+    dispatch({ type: 'hoverInIcon' });
   };
 
-  childRef: {| current: null | React.ElementRef<'div'> |} = React.createRef();
+  const handleIconMouseLeave = () => {
+    setTimeout(() => {
+      dispatch({ type: 'hoverOutIcon' });
+    }, 75);
```

Comment:
@anniejiwon Since we use the number `75` twice and they should be the same, could you extract it into a `const`?